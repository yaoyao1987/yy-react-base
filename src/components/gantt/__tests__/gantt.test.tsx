import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import GanttChart, { GanttChartProps } from '../gantt-chart'

const data = {
  info: {
    id: 49137,
    name: '新甘特图',
    timeUnit: 'D',
    sort: '1598518783044,1598518783047,1598518783050,1598518783053,1598518783055,1598518783058'
  },
  resources: [
    {
      id: 237077,
      nickname: '',
      name: '姚姚',
      systemId: 'HCWJ5WYABJ7H',
      photo:
        'http://thirdwx.qlogo.cn/mmopen/vi_32/Q0j4TwGTfTK8qkY7FUvM13ro0GLtfWq5cSrnDbea4n5MF67xTGC9tFwtOtkO6zY9xULCRqO5PypDEDn3DK8XkA/132',
      sex: 2,
    },
    {
      id: 237078,
      nickname: '张三',
      name: null,
      systemId: null,
      photo: null,
      sex: null,
    },
    {
      id: 237079,
      nickname: '李四',
      name: null,
      systemId: null,
      photo: null,
      sex: null,
    },
  ],
  links: [
    {
      id: 1598518783061,
      source: 1598518783047,
      target: 1598518783050,
      type: '0',
    },
    {
      id: 1598518783065,
      source: 1598518783050,
      target: 1598518783055,
      type: '0',
    },
    {
      id: 1598518783068,
      source: 1598518783055,
      target: 1598518783058,
      type: '0',
    },
  ],
  data: [
    {
      id: 1598518783044,
      text: '第一阶段',
      start_date: null,
      end_date: null,
      actual_start_date: null,
      actual_end_date: null,
      progress: 0,
      type: 'project',
      parent: 0,
      assigners: [],
      remark: null,
    },
    {
      id: 1598518783047,
      text: '任务一',
      start_date: '2020-08-22 08:59:43.000Z',
      end_date: '2020-08-25 08:59:43.000Z',
      actual_start_date: '2020-08-22 08:59:43.000Z',
      actual_end_date: '2020-08-25 08:59:43.000Z',
      progress: 1,
      type: 'task',
      parent: 1598518783044,
      assigners: ['237078'],
      remark: null,
    },
    {
      id: 1598518783050,
      text: '任务二',
      start_date: '2020-08-25 08:59:43.000Z',
      end_date: '2020-08-28 08:59:43.000Z',
      actual_start_date: '2020-08-25 08:59:43.000Z',
      actual_end_date: '2020-08-27 08:59:43.000Z',
      progress: 0.9,
      type: 'task',
      parent: 1598518783044,
      assigners: ['237079'],
      remark: null,
    },
    {
      id: 1598518783053,
      text: '第二阶段',
      start_date: null,
      end_date: null,
      actual_start_date: null,
      actual_end_date: null,
      progress: 0,
      type: 'project',
      parent: 0,
      assigners: [],
      remark: null,
    },
    {
      id: 1598518783055,
      text: '任务三',
      start_date: '2020-08-28 08:59:43.000Z',
      end_date: '2020-08-30 08:59:43.000Z',
      actual_start_date: null,
      actual_end_date: null,
      progress: 0,
      type: 'task',
      parent: 1598518783053,
      assigners: [],
      remark: null,
    },
    {
      id: 1598518783058,
      text: '里程碑',
      start_date: '2020-08-31 08:59:43.000Z',
      end_date: '2020-08-31 08:59:43.000Z',
      actual_start_date: null,
      actual_end_date: null,
      progress: 0,
      type: 'milestone',
      parent: 0,
      assigners: [],
      remark: null,
    },
  ],
}
const cols = [
  { view: 'grid', scrollX: 'scrollHor', scrollY: 'ganttScrollVer' },
  { resizer: true, width: 1 },
  { view: 'timeline', scrollX: 'scrollHor', scrollY: 'ganttScrollVer' },
  { view: 'scrollbar', id: 'ganttScrollVer' },
];

const columns = [
  { name: 'add', label: '', min_width: 40, max_width: 40 },
  {
    name: 'assigner',
    label: '负责人',
    align: 'left',
    min_width: 60,
    max_width: 60,
    template(task: any) {
      let temp = ''
      console.log('task.assigners', task.assigners)
      if (task.assigners) {
        task.assigners.forEach((item: string, index: number) => {
          temp += item
        })
      }
      return temp
    },
  },
  {
    name: 'text',
    label: '任务',
    width: '*',
    tree: true,
  },
  {
    name: 'start_date',
    label: '计划开始日',
    align: 'center',
    min_width: 80,
    max_width: 80,
  },
  {
    name: 'end_date',
    label: '计划结束日',
    align: 'center',
    min_width: 80,
    max_width: 80,
  },
  {
    name: 'actual_start_date',
    label: '实际开始日',
    align: 'center',
    min_width: 80,
    max_width: 80,
  },
  {
    name: 'actual_end_date',
    label: '实际结束日',
    align: 'center',
    min_width: 80,
    max_width: 80,
  },
  {
    name: 'progress',
    label: '进度',
    align: 'center',
    min_width: 50,
    max_width: 50,
  },
]

const testProps: GanttChartProps = {
  zoomLevel: "W",
  data,
  cols,
  columns
}
describe('test GanttChart component', () => {
  it('should render correctly', () => {
    const wrapper = render(<GanttChart zoomLevel="W" data={data} cols={cols} columns={columns} />)
    const element = wrapper.container.querySelector('.yy-gantt-container') as HTMLElement
    expect(element.tagName).toEqual('DIV')
    expect(element).toHaveClass('yy-gantt-container')
    expect(wrapper).toMatchSnapshot()
  });

  it('responses hover events', () => {
    const onMouseEnter = jest.fn();
    const onMouseLeave = jest.fn();
  });
})
