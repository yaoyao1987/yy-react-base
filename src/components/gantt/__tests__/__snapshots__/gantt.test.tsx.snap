// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GanttChart component should render correct component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="yy-gantt-container"
      >
        <div
          class="gantt_layout_cell gantt_layout_root gantt_layout gantt_layout_y gantt_container gantt_layout_cell_border_left gantt_layout_cell_border_top gantt_layout_cell_border_right gantt_layout_cell_border_bottom"
          data-cell-id="main"
          style="width: 0px; height: 0px; padding: 0px;"
        >
          <div
            class="gantt_layout_cell gantt_layout gantt_layout_x  gantt_layout_cell_border_transparent gantt_layout_cell_border_bottom"
            data-cell-id="c1598707613464"
            style="margin-bottom: 0px; width: -2px; height: 0px; padding: 0px;"
          >
            <div
              class="gantt_layout_cell  grid_cell gantt_layout_outer_scroll gantt_layout_outer_scroll_vertical gantt_layout_outer_scroll gantt_layout_outer_scroll_horizontal gantt_layout_cell_border_right"
              data-cell-id="1598707613465"
              style="margin-right: 0px; width: 540px; height: -1px;"
            >
              <div
                class="gantt_layout_content"
                style="height: -1px;"
              >
                <div
                  aria-multiselectable="false"
                  class="gantt_grid"
                  role="treegrid"
                  style="height:inherit;width:inherit;"
                >
                  <div
                    class="gantt_grid_scale"
                    role="row"
                    style="width: inherit; height: 35px; line-height: 34px;"
                  >
                    <div
                      aria-label="新任務"
                      class="gantt_grid_head_cell gantt_grid_head_add  "
                      column_id="add"
                      data-column-id="add"
                      role="button"
                      style="width:40px;"
                    />
                    <div
                      aria-label="负责人"
                      class="gantt_grid_head_cell gantt_grid_head_assigner  "
                      column_id="assigner"
                      data-column-id="assigner"
                      role="columnheader"
                      style="width:81px;"
                    >
                      负责人
                    </div>
                    <div
                      aria-label="任务"
                      class="gantt_grid_head_cell gantt_grid_head_text  "
                      column_id="text"
                      data-column-id="text"
                      role="columnheader"
                      style="width:70px;"
                    >
                      任务
                    </div>
                    <div
                      aria-label="计划开始日"
                      class="gantt_grid_head_cell gantt_grid_head_start_date  "
                      column_id="start_date"
                      data-column-id="start_date"
                      role="columnheader"
                      style="width:80px;"
                    >
                      计划开始日
                    </div>
                    <div
                      aria-label="计划结束日"
                      class="gantt_grid_head_cell gantt_grid_head_end_date  "
                      column_id="end_date"
                      data-column-id="end_date"
                      role="columnheader"
                      style="width:80px;"
                    >
                      计划结束日
                    </div>
                    <div
                      aria-label="实际开始日"
                      class="gantt_grid_head_cell gantt_grid_head_actual_start_date  "
                      column_id="actual_start_date"
                      data-column-id="actual_start_date"
                      role="columnheader"
                      style="width:80px;"
                    >
                      实际开始日
                    </div>
                    <div
                      aria-label="实际结束日"
                      class="gantt_grid_head_cell gantt_grid_head_actual_end_date  "
                      column_id="actual_end_date"
                      data-column-id="actual_end_date"
                      role="columnheader"
                      style="width:80px;"
                    >
                      实际结束日
                    </div>
                    <div
                      aria-label="进度"
                      class="gantt_grid_head_cell gantt_grid_head_progress gantt_last_cell "
                      column_id="progress"
                      data-column-id="progress"
                      role="columnheader"
                      style="width:49px;"
                    >
                      进度
                    </div>
                  </div>
                  <div
                    class="gantt_grid_data"
                    data-layer="true"
                    role="rowgroup"
                    style="width: inherit; height: 0px;"
                  >
                    <div
                      style="visibility: hidden; height: 212px; width: 1px;"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="gantt_layout_cell  timeline_cell gantt_layout_outer_scroll gantt_layout_outer_scroll_vertical gantt_layout_outer_scroll gantt_layout_outer_scroll_horizontal"
              data-cell-id="1598707613466"
              style="margin-right: 0px; width: 0px; height: -1px;"
            >
              <div
                class="gantt_layout_content"
                style="height: -1px;"
              >
                <div
                  class="gantt_task"
                  style="width:inherit;height:inherit;"
                >
                  <div
                    class="gantt_task_scale"
                    style="height: 35px; width: 840px;"
                  >
                    <div
                      class="gantt_scale_line"
                      style="height:34px;position:relative;line-height:34px"
                    >
                      <div
                        aria-label="21 8月"
                        class="gantt_scale_cell"
                        style="width:70px;height:34px;position:absolute;left:0px"
                      >
                        21 8月
                      </div>
                    </div>
                  </div>
                  <div
                    class="gantt_data_area"
                    style="height: -36px; width: 840px;"
                  >
                    <div
                      class="gantt_task_bg"
                      data-layer="true"
                      style="height: 210px; width: 840px;"
                    />
                    <div
                      class="gantt_links_area"
                      data-layer="true"
                      style="width: 840px;"
                    />
                    <div
                      class="gantt_bars_area"
                      data-layer="true"
                      style="width: 840px;"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="gantt_layout_cell  ganttScrollVer_cell"
              data-cell-id="1598707613467"
              style="width: 1px; height: -1px;"
            >
              <div
                class="gantt_layout_content gantt_task_vscroll"
                style="height: -1px;"
              >
                <div
                  class="gantt_layout_cell gantt_ver_scroll gantt_layout_cell_border_top"
                  style="width: 1px; height: -33px; top: 34px;"
                >
                  <div
                    style="height: 212px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="gantt_layout_cell  scrollHor_cell"
            data-cell-id="1598707613468"
            style="width: -2px; height: 1px;"
          >
            <div
              class="gantt_layout_content"
              style="height: 1px;"
            >
              <div
                class="gantt_layout_cell gantt_hor_scroll"
                style="width: -2px; height: 1px;"
              >
                <div
                  style="width: 840px;"
                />
              </div>
            </div>
          </div>
        </div>
        <iframe
          aria-hidden="true"
          class="gantt_container_resize_watcher"
          role="none"
          tabindex="-1"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="yy-gantt-container"
    >
      <div
        class="gantt_layout_cell gantt_layout_root gantt_layout gantt_layout_y gantt_container gantt_layout_cell_border_left gantt_layout_cell_border_top gantt_layout_cell_border_right gantt_layout_cell_border_bottom"
        data-cell-id="main"
        style="width: 0px; height: 0px; padding: 0px;"
      >
        <div
          class="gantt_layout_cell gantt_layout gantt_layout_x  gantt_layout_cell_border_transparent gantt_layout_cell_border_bottom"
          data-cell-id="c1598707613464"
          style="margin-bottom: 0px; width: -2px; height: 0px; padding: 0px;"
        >
          <div
            class="gantt_layout_cell  grid_cell gantt_layout_outer_scroll gantt_layout_outer_scroll_vertical gantt_layout_outer_scroll gantt_layout_outer_scroll_horizontal gantt_layout_cell_border_right"
            data-cell-id="1598707613465"
            style="margin-right: 0px; width: 540px; height: -1px;"
          >
            <div
              class="gantt_layout_content"
              style="height: -1px;"
            >
              <div
                aria-multiselectable="false"
                class="gantt_grid"
                role="treegrid"
                style="height:inherit;width:inherit;"
              >
                <div
                  class="gantt_grid_scale"
                  role="row"
                  style="width: inherit; height: 35px; line-height: 34px;"
                >
                  <div
                    aria-label="新任務"
                    class="gantt_grid_head_cell gantt_grid_head_add  "
                    column_id="add"
                    data-column-id="add"
                    role="button"
                    style="width:40px;"
                  />
                  <div
                    aria-label="负责人"
                    class="gantt_grid_head_cell gantt_grid_head_assigner  "
                    column_id="assigner"
                    data-column-id="assigner"
                    role="columnheader"
                    style="width:81px;"
                  >
                    负责人
                  </div>
                  <div
                    aria-label="任务"
                    class="gantt_grid_head_cell gantt_grid_head_text  "
                    column_id="text"
                    data-column-id="text"
                    role="columnheader"
                    style="width:70px;"
                  >
                    任务
                  </div>
                  <div
                    aria-label="计划开始日"
                    class="gantt_grid_head_cell gantt_grid_head_start_date  "
                    column_id="start_date"
                    data-column-id="start_date"
                    role="columnheader"
                    style="width:80px;"
                  >
                    计划开始日
                  </div>
                  <div
                    aria-label="计划结束日"
                    class="gantt_grid_head_cell gantt_grid_head_end_date  "
                    column_id="end_date"
                    data-column-id="end_date"
                    role="columnheader"
                    style="width:80px;"
                  >
                    计划结束日
                  </div>
                  <div
                    aria-label="实际开始日"
                    class="gantt_grid_head_cell gantt_grid_head_actual_start_date  "
                    column_id="actual_start_date"
                    data-column-id="actual_start_date"
                    role="columnheader"
                    style="width:80px;"
                  >
                    实际开始日
                  </div>
                  <div
                    aria-label="实际结束日"
                    class="gantt_grid_head_cell gantt_grid_head_actual_end_date  "
                    column_id="actual_end_date"
                    data-column-id="actual_end_date"
                    role="columnheader"
                    style="width:80px;"
                  >
                    实际结束日
                  </div>
                  <div
                    aria-label="进度"
                    class="gantt_grid_head_cell gantt_grid_head_progress gantt_last_cell "
                    column_id="progress"
                    data-column-id="progress"
                    role="columnheader"
                    style="width:49px;"
                  >
                    进度
                  </div>
                </div>
                <div
                  class="gantt_grid_data"
                  data-layer="true"
                  role="rowgroup"
                  style="width: inherit; height: 0px;"
                >
                  <div
                    style="visibility: hidden; height: 212px; width: 1px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="gantt_layout_cell  timeline_cell gantt_layout_outer_scroll gantt_layout_outer_scroll_vertical gantt_layout_outer_scroll gantt_layout_outer_scroll_horizontal"
            data-cell-id="1598707613466"
            style="margin-right: 0px; width: 0px; height: -1px;"
          >
            <div
              class="gantt_layout_content"
              style="height: -1px;"
            >
              <div
                class="gantt_task"
                style="width:inherit;height:inherit;"
              >
                <div
                  class="gantt_task_scale"
                  style="height: 35px; width: 840px;"
                >
                  <div
                    class="gantt_scale_line"
                    style="height:34px;position:relative;line-height:34px"
                  >
                    <div
                      aria-label="21 8月"
                      class="gantt_scale_cell"
                      style="width:70px;height:34px;position:absolute;left:0px"
                    >
                      21 8月
                    </div>
                  </div>
                </div>
                <div
                  class="gantt_data_area"
                  style="height: -36px; width: 840px;"
                >
                  <div
                    class="gantt_task_bg"
                    data-layer="true"
                    style="height: 210px; width: 840px;"
                  />
                  <div
                    class="gantt_links_area"
                    data-layer="true"
                    style="width: 840px;"
                  />
                  <div
                    class="gantt_bars_area"
                    data-layer="true"
                    style="width: 840px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="gantt_layout_cell  ganttScrollVer_cell"
            data-cell-id="1598707613467"
            style="width: 1px; height: -1px;"
          >
            <div
              class="gantt_layout_content gantt_task_vscroll"
              style="height: -1px;"
            >
              <div
                class="gantt_layout_cell gantt_ver_scroll gantt_layout_cell_border_top"
                style="width: 1px; height: -33px; top: 34px;"
              >
                <div
                  style="height: 212px;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="gantt_layout_cell  scrollHor_cell"
          data-cell-id="1598707613468"
          style="width: -2px; height: 1px;"
        >
          <div
            class="gantt_layout_content"
            style="height: 1px;"
          >
            <div
              class="gantt_layout_cell gantt_hor_scroll"
              style="width: -2px; height: 1px;"
            >
              <div
                style="width: 840px;"
              />
            </div>
          </div>
        </div>
      </div>
      <iframe
        aria-hidden="true"
        class="gantt_container_resize_watcher"
        role="none"
        tabindex="-1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
